pkgs/applications/video/mpv/default.nix

```diff
@@ -100,6 +100,10 @@ in stdenv.mkDerivation rec {
   NIX_LDFLAGS = lib.optionalString x11Support "-lX11 -lXext "
     + lib.optionalString stdenv.isDarwin "-framework CoreFoundation";
 
+  # These flags are not supported and cause the build
+  # to fail, even when cross compilation itself works.
+  dontAddWafCrossFlags = true;
+
```

Comment:
wayland-scanner has to be added to `nativeBuildInputs` when wayland is enabled

Reply:
I can build with wayland support and it apparently works. Here's a screenshot of mpv compiled for armv6 running in weston (under X.org):

![image](https://user-images.githubusercontent.com/2817565/187040835-2229a8db-eb58-42d1-bd33-5b59dfe10fe4.png)